app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -636,7 +635,7 @@ fun CollapsibleHikesList(
                   CircularProgressIndicator(
                       modifier = Modifier.testTag(MapScreen.TEST_TAG_SEARCH_LOADING_ANIMATION))
                 }
-          } else if (routes.value.isEmpty()) {
+          } else if (hikes.isEmpty() || hikesType != HikesViewModel.LoadedHikes.FromBounds) {
```

Comment:
Wouldn't that be better to either
- Remove the loaded hikes when switching a tab
- Remove the loaded hikes when going to the map screen

How is this handled from map -> saved hikes ?

Reply:
If we remove the hikes, we need to re-compute them, which is not the case if we keep them without displaying.

When you go back to the map, the hikes list is not cleared, it contains the saved hikes. Once you search for hikes, the list will be cleared, except for the hikes that are already loaded. Those ones will be kept and already have all of their data.

When going from map screen to saved hikes, the same happens, but the list is cleared right away because it loads saved hikes right away.

I could add a function to the view model to clear the hikes list, but I'm not sure it's necessary.

Reply:
Thanks for the explanation. It's fine like this